File: samples/CS/08-Images/RibbonItems.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Drawing;
using System.Windows.Forms;

namespace WinForms.Ribbon
{
    partial class RibbonItems
    {
        private bool exitOn = false;

        public void Init()
        {
            ButtonDropA.Click += new EventHandler<EventArgs>(_buttonDropA_ExecuteEvent);
            ButtonDropB.Click += new EventHandler<EventArgs>(_buttonDropB_ExecuteEvent);
        }

        unsafe void _buttonDropA_ExecuteEvent(object sender, EventArgs e)
        {
            // load bitmap from file
            Bitmap bitmap = new System.Drawing.Bitmap(@"..\..\..\Res\Drop32.bmp");
            //bitmap.MakeTransparent();

            // set large image property
            ButtonDropA.LargeImage = new UIImage(Ribbon, bitmap).UIImageHandle; //.ConvertToUIImage(bitmap);
        }

        unsafe void _buttonDropB_ExecuteEvent(object sender, EventArgs e)
        {
            List<int> supportedImageSizes = new List<int>() { 32, 48, 64 };

            Bitmap bitmap;
            StringBuilder bitmapFileName = new StringBuilder();

            int selectedImageSize;
            if (supportedImageSizes.Contains(SystemInformation.IconSize.Width))
            {
                selectedImageSize = SystemInformation.IconSize.Width;
            }
            else
            {
                selectedImageSize = 32;
            }

            exitOn = !exitOn;
            string exitStatus = exitOn ? "on" : "off";

            bitmapFileName.AppendFormat(@"..\..\..\Res\Exit{0}{1}.bmp", exitStatus, selectedImageSize);

            bitmap = new System.Drawing.Bitmap(bitmapFileName.ToString());
            bitmap.MakeTransparent();

            ButtonDropB.LargeImage = new UIImage(Ribbon, bitmap).UIImageHandle; //.ConvertToUIImage(bitmap);
        }

        public void Load()
        {
        }

    }
}

File: samples/VB/08-Images/RibbonItems.vb
Imports System
Imports System.Text

Namespace WinForms.Ribbon

    Partial Class RibbonItems
		Private exitOn As Boolean = False

		Public Sub Init()
			AddHandler _ButtonDropA.Click, AddressOf _buttonDropA_ExecuteEvent
			AddHandler _ButtonDropB.Click, AddressOf _buttonDropB_ExecuteEvent
		End Sub

		Private Sub _buttonDropA_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			' load bitmap from file
			Dim bitmap_Renamed As Bitmap = New Bitmap("..\..\..\Res\Drop32.bmp")
			'bitmap_Renamed.MakeTransparent()

			' set large image property
			_ButtonDropA.LargeImage = New UIImage(Ribbon, bitmap_Renamed)
		End Sub

		Private Sub _buttonDropB_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			Dim supportedImageSizes As New List(Of Integer)() From {32, 48, 64}

			Dim bitmap_Renamed As Bitmap
			Dim bitmapFileName As New StringBuilder()

			Dim selectedImageSize As Integer
			If supportedImageSizes.Contains(SystemInformation.IconSize.Width) Then
				selectedImageSize = SystemInformation.IconSize.Width
			Else
				selectedImageSize = 32
			End If

			exitOn = Not exitOn
			Dim exitStatus As String = If(exitOn, "on", "off")

			bitmapFileName.AppendFormat("..\..\..\Res\Exit{0}{1}.bmp", exitStatus, selectedImageSize)

			bitmap_Renamed = New Bitmap(bitmapFileName.ToString())
			'bitmap_Renamed.MakeTransparent()

			_ButtonDropB.LargeImage = New UIImage(Ribbon, bitmap_Renamed)
		End Sub

	End Class
End Namespace
